packages/swapper/src/swappers/ZrxSwapper/getZrxTradeRate/getZrxTradeRate.ts

```diff
@@ -242,9 +242,14 @@ async function _getZrxPermit2TradeRate(
     totalNetworkFee,
   } = zrxPriceResponse
 
-  const rate = bnOrZero(buyAmountAfterFeesCryptoBaseUnit)
-    .div(sellAmountIncludingProtocolFeesCryptoBaseUnit)
-    .toString()
+  // for the rate to be valid, both amounts must be converted to the same precision
+  const rate = convertPrecision({
+    value: buyAmountAfterFeesCryptoBaseUnit,
+    inputExponent: buyAsset.precision,
+    outputExponent: sellAsset.precision,
+  })
+    .dividedBy(bn(sellAmountIncludingProtocolFeesCryptoBaseUnit))
```

Comment:
[nitpick]: bignumber.js doesnt require explicit conversion into a `bn`, but I aint gonna block this.
```suggestion
    .dividedBy(sellAmountIncludingProtocolFeesCryptoBaseUnit)
```